pkg/api/common/versions.go

```diff
@@ -133,18 +127,13 @@ func GetVersionsGt(versions []string, version string, inclusive, preReleases boo
 func GetVersionsLt(versions []string, version string, inclusive, preReleases bool) []string {
 	// Try to get latest version matching the release
 	var ret []string
-	var cons *semver.Constraints
+	minVersion, _ := semver.Make(version)
 	for _, v := range versions {
-		sv, _ := semver.NewVersion(v)
-		if preReleases {
-			sv, _ = semver.NewVersion(fmt.Sprintf("%d.%d.%d", sv.Major(), sv.Minor(), sv.Patch()))
-		}
-		if inclusive {
-			cons, _ = semver.NewConstraint("<=" + version)
-		} else {
-			cons, _ = semver.NewConstraint("<" + version)
+		sv, _ := semver.Make(v)
+		if !preReleases && len(sv.Pre) != 0 {
```

Comment:
if `preReleases` is false we don't want to include pre-releases so we skip any version with a pre-release part